### PR TITLE
Refactor(web, web-twig, web-react): FileUploader imageObjectFit from data-attr to CSS

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

- Update the implementation to set image object fit directly using CSS properties, eliminating the need for the data attribute
- Added demo for FileUploaderAttachment with image-object-fit: contain in all packages

### Additional context

It is now working without the need for a JS plugin

### Demo
image

### Issue reference

[Object fit nefunguje ve Twigu](https://jira.lmc.cz/browse/DS-1036)

### CODE REVIEW

File: packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig:1]: Please update snapshot tests so we can be sure it works.

At minimum, it should be in the "Render with all props" section at the end.

File: packages/web-react/src/components/FileUploader/demo/FileUploaderAttachment.tsx [hunk 1]
```diff
@@ -30,6 +31,15 @@ const FileUploaderAttachment = () => {
...
           }}
         />
+        <FileUploaderAttachmentComponent
+          hasImagePreview
+          name="test"
+          file={new File([byteArrayPortrait], 'test1.png', { type: 'image/png', lastModified: 123456789 })}
+          id="2"
+          label="Contained image"
+          onDismiss={onDismiss}
+          imageObjectFit="contain"
```
Comment (adamkudrna) [inline packages/web-react/src/components/FileUploader/demo/FileUploaderAttachment.tsx:41]: It seems we do not have a unit test for this property, or do we?

File: packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig [hunk 2]
```diff
@@ -19,7 +19,8 @@
 
 {# Attributes #}
 {%- set _dataImagePreviewAttr = _generateImagePreview ? 'data-spirit-imagePreview="true"' : null -%}
-{%- set _imageObjectFitAttr = _imageObjectFit ? 'data-spirit-image-object-fit=' ~ _imageObjectFit : null -%}
+{%- set _imageObjectFitCSSAttr = '--file-uploader-attachment-image-object-fit:' -%}
+{%- set _imageObjectFitAttr = _imageObjectFit ? 'style=' ~ _imageObjectFitCSSAttr ~ _imageObjectFit : null -%}
```
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig:23]: The `_imageObjectFitCSSAttr` variable is not necessary. This is how we deal with unwanted escaping in our Twig components:

```suggestion
{%- set _imageObjectFitAttr = _imageObjectFit ? 'style="--file-uploader-attachment-image-object-fit: ' ~ _imageObjectFit | escape('html_attr') ~ '"' : null -%}
```

Then you need to call the `raw` filter on it and it's done:

```twig
{{ _imageObjectFitAttr | raw }}
```

Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig:23]: If Prettier allows, the line with the image (out of scope of this PR) could be rather expanded to multiple lines:

```twig
<img
  src="{{ _imagePreview }}"
  {{ _imageObjectFitAttr | raw }}
  width="54"
  height="54"
  alt="{{- _fileName -}}"
/>
```

Comment (pavelklibani) [reply packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig]: Thank you! Fixed 🪄

Comment (pavelklibani) [reply packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig]: done

File: packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig
Comment (pavelklibani) [reply packages/web-twig/src/Resources/components/FileUploader/FileUploaderAttachment.twig:1]: test added

File: packages/web-react/src/components/FileUploader/demo/FileUploaderAttachment.tsx (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/FileUploader/demo/FileUploaderAttachment.tsx:41]: test added

File: packages/web/src/js/FileUploader.ts [hunk 3]
```diff
@@ -265,10 +265,8 @@ class FileUploader extends BaseComponent {
     const itemButton = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="button"]`);
     const hasImagePreview = Boolean((item as HTMLElement)?.dataset?.spiritImagepreview);
     const AttachmentSvgIcon = item?.querySelector('svg');
-    const AttachmentPreviewImage = snippet.querySelector('.FileUploaderAttachment__image');
+    const AttachmentPreviewImage = item?.querySelector('[data-spirit-element="attachment-image"]');
```
Comment (literat) [inline packages/web/src/js/FileUploader.ts:268]: You are repeating a `[data-spirit-element="attachment-image"]` selector, so you can make a constant from this value and use it on both places.

**Review by literat — APPROVED** [review]
Nice work 👍

File: packages/web/src/js/FileUploader.ts (hunk 3 above)
Comment (pavelklibani) [reply packages/web/src/js/FileUploader.ts:268]: i'll do that, thank you

**Pull request merged**